src/ui/gum-handler.html

```diff
@@ -78,25 +80,23 @@
           }
         }
       }
-      this.getUserMedia(constraints, this.gotStream_.bind(this), this.gotError_.bind(this));
-    },
-
-    gotStream_: function(stream) {
-      // Stop all tracks to ensure the camera and audio devices are shutdown
-      // directly.
-      for (var i = 0; i < stream.getVideoTracks().length; i++) {
-        stream.getVideoTracks()[i].stop();
-      }
-      for (var j = 0; j < stream.getAudioTracks().length; j++) {
-        stream.getAudioTracks()[j].stop();
-      }
-      this.pending = false;
-      this.error = false;
+      this.getUserMedia(constraints)
+      .then(function(stream) {
+        // Stop all tracks to ensure the camera and audio devices are shutdown
+        // directly.
+        for (var i = 0; i < stream.getVideoTracks().length; i++) {
+          stream.getVideoTracks()[i].stop();
+        }
+        for (var j = 0; j < stream.getAudioTracks().length; j++) {
+          stream.getAudioTracks()[j].stop();
+        }
+        this.pending = false;
+        this.error = false;
+      }.bind(this))
```

Comment:
This syntax looks weird:
 1 - not sure about indentation, I would expect .then/.catch to be indented as is common on fluent apis.
 2 - missing parenthesis around (function() {...})  before calling bind(). E:g. lalal(function() {}.bind(...)); looks very weird to me.